packages/destination-actions/src/destinations/delivrai-activate/updateSegment/index.ts

```diff
@@ -0,0 +1,153 @@
+      unsafe_hidden: false,
+      required: false,
+      default: {
+        '@path': '$.context.personas.computation_id'
+      }
+    },
+    email: {
+      label: 'User Email',
+      description: 'Email address of a user',
+      type: 'string',
+      unsafe_hidden: false,
+      required: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.email' },
+          then: { '@path': '$.traits.email' },
+          else: { '@path': '$.context.traits.email' } // Phone is sent as identify's trait or track's context.trait
+        }
+      }
+    },
+    phone: {
+      label: 'User Phone',
+      description: 'Phone number of a user',
+      type: 'string',
+      unsafe_hidden: false,
+      required: false,
+      default: {
+        '@if': {
+          exists: { '@path': '$.traits.phone' },
+          then: { '@path': '$.traits.phone' }, // Phone is sent as identify's trait or track's property
+          else: { '@path': '$.properties.phone' }
+        }
+      }
+    },
+    advertising_id: {
+      label: 'User Mobile Advertising ID',
+      description: "User's mobile advertising Id",
+      type: 'string',
+      unsafe_hidden: false,
+      required: false,
+      default: {
+        '@path': '$.context.device.advertisingId'
+      }
+    },
+    segment_audience_key: {
+      label: 'Segment Audience Key',
+      description: 'Segment Audience Key. Maps to the "Name" of the Segment node in Delivr AI Audience Segmentation',
+      type: 'string',
+      // unsafe_hidden: true,
+      required: true,
+      default: {
+        '@path': '$.context.personas.computation_key'
+      }
+    },
+    enable_batching: {
+      label: 'Batch Data to Delivr AI',
+      description:
+        'If true, batch requests to Delivr AI. Delivr AI accepts batches of up to 1000 events. If false, send each event individually.',
+      type: 'boolean',
+      default: true,
+      unsafe_hidden: true
+    },
+    batch_size: {
+      label: 'Batch Size',
+      description: 'Maximum number of events to include in each batch. Actual batch sizes may be lower.',
+      type: 'number',
+      default: 1000,
+      unsafe_hidden: true
+    }
+  },
+
+  perform: async (request, { payload, settings }) => {
+      // Check if client_identifier_id is valid
+      if (!settings?.client_identifier_id) {
+          return [{ status: false, message: "Client identifier is required" }];
+      }
+
+      // Generate JWT token
+      const rt_access_token_response = await generate_jwt(settings.client_identifier_id, request);
+      
+      // Check if token generation was successful
+      if (rt_access_token_response === 401) {
+          return [{ status: false, message: "Not valid customer" }];
+      }
+
+      const rt_access_token = rt_access_token_response?.data?.token;
+
+      // Process the payload with the valid token
+      return process_payload(request, [payload], rt_access_token, settings.client_identifier_id);
```

Comment:
```suggestion
    let rt_access_token_response
    // Generate JWT token
    try {
      rt_access_token_response = await generate_jwt(settings.client_identifier_id, request)
    } catch (error) {
      if (error === 401) {
        throw new InvalidAuthenticationError('Invalid Client Identifier ID')
      } else {
        throw new APIError('Error Generating JWT', 400)
      }
    }
    const rt_access_token = rt_access_token_response?.data?.token
    // Process the payload with the valid token
    return process_payload(request, [payload], rt_access_token, settings.client_identifier_id)
```

You can get rid of the settings condition as a user cannot enable the destination without providing a required setting like `client_identifier_id`. 

The suggestion above also standardized the error handling for `generate_jwt` and uses the built in Segment errors for better observability